crates/wasmtime/src/runtime/store.rs

```diff
@@ -260,6 +260,15 @@ pub enum UpdateDeadline {
     /// configured via [`Config::async_support`](crate::Config::async_support).
     #[cfg(feature = "async")]
     Yield(u64),
+    /// Extend the deadline by the specified number of ticks after yielding to
+    /// the async executor loop. This can only be used with an async [`Store`]
+    /// configured via [`Config::async_support`](crate::Config::async_support).
+    ///
+    /// The yield will be performed by the future provided; when using `tokio`
+    /// it is recommended to provide [`tokio::task::yield_now`](https://docs.rs/tokio/latest/tokio/task/fn.yield_now.html)
+    /// here.
+    #[cfg(feature = "async")]
+    YieldCustom(u64, Box<dyn ::core::future::Future<Output = ()> + Send>),
```

Comment:
Drive-by-comment on this unrelated to this change, mind throwing `#[non_exhaustive]` on this enum definition? That'll help consumers work with this a bit more nicely with the `#[cfg]`'d variants